packages/core/src/FloatingActionButtonProvider/useFloatingActionButton.ts

```diff
@@ -0,0 +1,60 @@
+import { useState, useEffect } from 'react'
+import { getScrollPercentage, delay, addScrollListener } from './helpers'
+
+const useFloatingActinButton = ({ hideUntilScrollPercent, delayDisplayMs }) => {
+  const displayOnLoad = hideUntilScrollPercent ? false : true
+  const [showFab, setShowFab] = useState(displayOnLoad)
+  const [showWrapper, setShowWrapper] = useState(displayOnLoad)
+  const [isDelayedDisplay, setIsDelayedDisplay] = useState(delayDisplayMs ? true : false)
+
+  const enterAnimation = async () => {
+    setShowFab(true)
+    setShowWrapper(true)
+  }
+
+  const exitAnimation = async () => {
+    setShowFab(false)
+    await delay(100)
+    setShowWrapper(false)
+  }
+
+  const handleScroll = async () =>
+    getScrollPercentage() > hideUntilScrollPercent
+      ? enterAnimation()
+      : exitAnimation()
+          .then(() => {
+            //Do Nothing
+          })
+          .catch(() => {
+            //Do nothing
+          })
+
+  const delayDisplay = async () =>
+    delay(delayDisplayMs)
+      .then(() => {
+        setIsDelayedDisplay(false)
+      })
+      .catch(() => {
+        //Do nothing
+      })
+
+  useEffect(() => typeof window !== 'undefined' && hideUntilScrollPercent && addScrollListener(handleScroll))
+
+  useEffect(() => {
+    delayDisplay()
+      .then(() => {
+        //Do nothing
+      })
+      .catch(() => {
+        //Do nothing
+      })
```

Comment:
if these don't do anything do we need them?

Reply:
@James300, I was getting linting errors when I had await, until I added these

Reply:
Code changed after using AnimatePresence